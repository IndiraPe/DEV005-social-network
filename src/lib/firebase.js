// Import the functions you need from the SDKs you need
import { initializeApp } from 'firebase/app';
import { getAuth } from 'firebase/auth';
import {
  getFirestore, doc, deleteDoc, collection, addDoc, getDocs, onSnapshot, orderBy, updateDoc, getDoc,
} from 'firebase/firestore';

// TODO: Add SDKs for Firebase products that you want to use
// https://firebase.google.com/docs/web/setup#available-libraries

// Your web app's Firebase configuration
// const firebaseConfig = {
//   apiKey: 'AIzaSyDr_i_8XfqSbS2ej-v0w-r6Wy2FCpmL_sw',
//   authDomain: 'social-network-b08d0.firebaseapp.com',
//   projectId: 'social-network-b08d0',
//   storageBucket: 'social-network-b08d0.appspot.com',
//   messagingSenderId: '611309449971',
//   appId: '1:611309449971:web:52cdfd8a7f8dfccac87d86',
// };
const firebaseConfig = {
  apiKey: 'AIzaSyBpUAmyGv1olj6CsTJ6A8rZtCvK3XgzpXA',
  authDomain: 'myfirstproyect-bb58e.firebaseapp.com',
  projectId: 'myfirstproyect-bb58e',
  storageBucket: 'myfirstproyect-bb58e.appspot.com',
  messagingSenderId: '849896345504',
  appId: '1:849896345504:web:4482e1224f5e61429d5ec5',
  measurementId: 'G-5HVRKB5KL9',
};
// Initialize Firebase
export const app = initializeApp(firebaseConfig);
export const auth = getAuth(app);
// Initialize Cloud Firestore and get a reference to the service
export const db = getFirestore(app);
// FunciÃ³n para guardar posts
export const savePost = (idUser, post, date) => {
  addDoc(collection(db, 'posts'), { idUser, post, date });
};

export const getPosts = () => getDocs(collection(db, 'posts'));

export const onGetPosts = (callback) => onSnapshot(collection(db, 'posts'), orderBy('datePost', 'des'), callback);

export const deletePost = (id) => deleteDoc(doc(db, 'posts', id));
export const getPost = (id) => getDoc(doc(db, 'posts', id));
export const editPost = (id, newPost) => updateDoc(doc(db, 'posts', id), newPost);
