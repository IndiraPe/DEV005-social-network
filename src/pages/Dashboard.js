import { buttonSignOut } from '../lib/LoginGoogle';

const Dashboard = (navigateTo) => {
  const viewDashboard = `
    <div id="headerDashboard">
      <h1 class="tittle-dashboard">Frikis Del Espacio</h1>
      <article id= "LogOut">
        <img id="logoSignOut" src= "./img/vaca2.png" alt="Imagen  "/>
        <button id ="button-signOut">Cerrar Sesión</button>
      </article>
    </div>
    <div id="barDashboard"></div>
    <div id="circleAlien">
      <div id= "circleAlien2">
        <img id="imgAlien" src= "./img/imgAlienDashboard.png" alt="Imagen de Alien "/>
      </div>
    </div>
    <p><span id= "userName"> Aquí va el nombre del usuario.</span></p>
    <div id = "border-textArea">
    
    <textarea id="post-text" name="textarea" placeholder="Escribir publicación..."></textarea>
    </div>
   <button id="button-post" class="btn-post">Continúa con Google
    </button>

    <div id='containerPosts'></div>
    `;

  const mainDashboard = document.createElement('div');
  mainDashboard.classList.add('main-dashboard');
  mainDashboard.innerHTML = viewDashboard;

  const buttonOut = mainDashboard.querySelector('#button-signOut');
  buttonSignOut(buttonOut, navigateTo);
  return mainDashboard;
};
export default Dashboard;
