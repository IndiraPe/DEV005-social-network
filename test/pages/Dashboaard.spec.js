/**
 * @jest-environment jsdom
 */
import * as FIREBASE from 'firebase/auth';
import Dashboard from '../../src/pages/Dashboard.js';

jest.mock('firebase/auth');

describe('Dashboard', () => {
  it('Debería ser una función', () => {
    expect(typeof Dashboard).toBe('function');
  });
  it('Have a buttom', () => {
    FIREBASE.onAuthStateChanged.mockImplementation(() => {

    });
    const DOM = document.createElement('div');
    DOM.append(Dashboard());
    const haveAbutton = DOM.querySelector('#button-signOut');
    expect(haveAbutton).not.toBe(undefined);
  });
  test('After click button signOut return call function navigateTo', (done) => {
    FIREBASE.signOut.mockImplementation(() => Promise.resolve());

    const DOM = document.createElement('div');
    const navigateTo = jest.fn();
    DOM.append(Dashboard(navigateTo));
    const buttonSignOut = DOM.querySelector('#button-signOut');
    buttonSignOut.click();
    // expect(navigateTo).toHaveBeenCalledTimes(1);
    setTimeout(() => {
      expect(navigateTo).toHaveBeenLastCalledWith('/');
      done();
    }, 0);
  });
  test('snapshot of Dashboard', () => {
    const DOM = document.createElement('div');
    DOM.append(Dashboard());
    expect(DOM).toMatchSnapshot();
  });
});
