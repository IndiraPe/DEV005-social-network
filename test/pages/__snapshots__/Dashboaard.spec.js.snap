// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard snapshot of Dashboard 1`] = `
<div>
  <div
    class="main-dashboard"
  >
    
    
    <div
      id="headerDashboard"
    >
      
      
      <h1
        class="tittle-dashboard"
      >
        Frikis Del Espacio
      </h1>
      
      
      <article
        id="LogOut"
      >
        
        
        <img
          alt="Imagen  "
          id="logoSignOut"
          src="./img/vaca2.png"
        />
        
        
        <button
          id="button-signOut"
        >
          Cerrar Sesión
        </button>
        
      
      </article>
      
    
    </div>
    
    
    <div
      id="barDashboard"
    />
    
    
    
    <div
      id="profile"
    >
      
    
      <div
        id="circleAlien"
      >
        
      
        <div
          id="circleAlien2"
        >
          
        
          <img
            alt="Imagen de Alien "
            id="imgAlien"
            src="./img/imgAlienDashboard.png"
          />
          
      
        </div>
        
    
      </div>
      
    
      <span
        id="userName"
      >
         Gabriela79 
      </span>
      
    
    </div>
    
    
    <div
      id="border-textArea"
    >
      
      
      <textarea
        id="post-text"
        name="textarea"
        placeholder="Escribir publicación..."
      />
      
      
    
    </div>
    
    
    <div>
      
    
      <input
        id="img-post"
        msg-post"=""
        type="file"
        value="Subir Imágen >
      </div>
    <div id = "
      />
      
      
      <span
        id="msg-post-text"
      >
        Se ha compartido tu publicación
      </span>
      
    
    </div>
    
    
    <div
      id="container-btn"
    >
      
      
      <button
        class="btn-post"
        id="button-post"
      >
        Publicar
      </button>
      
    
    </div>
    

    
    <div
      id="containerPosts"
    >
      
      
      <div
        class="box-gradient"
      >
        
        
        <div
          id="postPublic"
        >
          
          
          <span
            class="loading"
          >
            Cargando...
          </span>
          
          
          <img
            alt="Ovni girando"
            id="ovniL"
            src="./img/ovni.gif"
          />
          
        
        </div>
        
      
      </div>
          
    
    </div>
    
    
    <div
      class="modal"
      id="myModal"
    >
      
      
      <div
        class="modal-content"
      >
        
        
        <p>
          ¿Estás seguro(a) de eliminar tu publicación?
        </p>
        
        
        <button
          id="yes"
        >
          Sí
        </button>
        
        
        <button
          id="not"
        >
          No
        </button>
        
      
      </div>
      
    
    </div>
    
    
  </div>
</div>
`;
